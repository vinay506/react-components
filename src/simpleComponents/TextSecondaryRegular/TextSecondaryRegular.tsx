import React from 'react'
import styled from 'styled-components';
function TextSecondaryRegular(props: any) {
    /**
     * Text: The text which should be displayed on web page
     * color: Defaults to black, To override the existing color we can pass Color code or color name as props.
     */
    const { text, color } = props;
    const customStyles = {
        color: color
    };
    return (
        <React.Fragment>
            <Label id="textSecondaryRegular" style={customStyles}>
                {text}
            </Label>
        </React.Fragment >
    )
}

const Label = styled.div`
    font-size: 18px;
    color: black;
    font-family: 'AmazonEmber-regular';
    letterSpacing: 0;
    lineHeight: 24px;
    `;

export default TextSecondaryRegular
