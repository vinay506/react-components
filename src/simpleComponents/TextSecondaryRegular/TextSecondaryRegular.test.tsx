import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import 'jest-styled-components';
import TextSecondaryRegular from './TextSecondaryRegular';
describe('Text Secondary', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles: any;
    beforeEach(() => {
        propInfo = {
            text: 'Sample Text Secondary',
            color: 'red'
        };
        defaultStyles = {
            color: 'black',
            fontSize: '18px',
            fontFamily: 'AmazonEmber-regular',
            letterSpacing: '0',
            lineHeight: '24px'

        }
        wrapper = shallow(<TextSecondaryRegular {...propInfo} />);
    });
    test('Secondary Text Snapshot', () => {
        const instance = renderer.create(<TextSecondaryRegular {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Secondary Text Element check', () => {
        const elementLength = wrapper.find('#textSecondaryRegular').length;
        expect(elementLength).toEqual(1);
    })
    test('Secondary Text Validation', () => {
        const primaryText = wrapper.find('#textSecondaryRegular').text();
        expect(primaryText).toEqual(propInfo.text);
    })
    test('Secondary Text Color Validation', () => {
        const elementStyles = wrapper.find('#textSecondaryRegular');
        expect(elementStyles.props().style.color).toEqual(propInfo.color);
    })

    test('Text Secondary Default Color Validation', () => {
        const instance = renderer.create(<TextSecondaryRegular {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.color);
    })

    test('Text Secondary Default font-size Validation', () => {
        const instance = renderer.create(<TextSecondaryRegular {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontSize);
    })

    test('Text Secondary Default font-family Validation', () => {
        const instance = renderer.create(<TextSecondaryRegular {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontFamily);
    })

    test('Text Secondary Default letter Spacing Validation', () => {
        const instance = renderer.create(<TextSecondaryRegular {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.letterSpacing);
    })

    test('Text Secondary Default line Height Validation', () => {
        const instance = renderer.create(<TextSecondaryRegular {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.lineHeight);
    })
})

