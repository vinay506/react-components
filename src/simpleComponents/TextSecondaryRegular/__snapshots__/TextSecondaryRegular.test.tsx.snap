// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Secondary Secondary Text Snapshot 1`] = `
.c0 {
  font-size: 18px;
  color: black;
  font-family: 'AmazonEmber-regular';
  -webkit-letterSpacing: 0;
  -moz-letterSpacing: 0;
  -ms-letterSpacing: 0;
  letterSpacing: 0;
  lineHeight: 24px;
}

<div
  className="c0"
  id="textSecondaryRegular"
  style={
    Object {
      "color": "red",
    }
  }
>
  Sample Text Secondary
</div>
`;
